tensorboard/plugins/hparams/api.proto

```diff
@@ -0,0 +1,187 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// Defines a proto3-based REST API that the HParams web-component of the plugin
+// would use to read data from a hyperparameter-tuning experiment.
+// The first part of this file defines the message types (resources) used
+// to pass information into and out of the API methods. These messages will be
+// transmited using their JSON encoding. The second part documents the actual
+// API and HTTP end-points.
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package tensorboard.plugins.hparams;
+
+// General note: in what follows we use the the field 'name' of a message to
+// stores its id. We avoid calling this field 'id' since it is a reserved word
+// in python and to be compliant with the API style guide detailed in
+// https://cloud.google.com/apis/design/.
+
+
+// Represents a single experiment.
+// An experiment consists of multiple "sessions". Typically, in each session
+// a model is trained for a given set of hyperparameter values. In each session
+// a training program may generate one or more series of real
+// numbers--each containing the evaluation of some metric at different
+// training steps.
+//
+// In some experiments, the user trains a model with the same set of
+// hyperparameters multiple times to get the distribution of metric
+// evaluations when the computation (such as the training algorithm, or metric
+// evaluation) is non-deterministic. To make the UI aware of this, sessions
+// are partitioned into groups: each group consists of all training sessions
+// which share the same values for the hyperparameters. In experiments with no
+// repeated executions, each group consists of exactly one session.
+message Experiment {
+  // A unique id.
+  string name = 1;
+
+  string description = 2;
+
+  // An id for the owning user or group.
+  string user = 3;
+
+  google.protobuf.Timestamp time_created = 4;
+
+  // Information about each hyperparameter used in the experiment.
+  repeated HParamInfo hparam_infos = 5;
+
+  // Information about each metric used in the experiment.
+  repeated MetricInfo metric_infos = 6;
+}
+
+message HParamInfo {
+  // An id for the hyperparameter (unique within an experiment).
+  string name = 1;
+
+  // A string used to display the hyperparameter in the UI. If empty the UI
+  // will display the 'name' field.
+  string display_name = 2;
+
+  string description = 3;
+
+  // The list of distinct values this hyperparameter had in any session within
+  // the experiment. Currently, the UI considers hyperparameter values as opaque
+  // strings.
+  repeated string vals = 4;
```

Comment:
Is this repeated to be able to specify multidimensional hyperparameters?
If so, please clarify that in the comment.

Reply:
I've changed this and added clarifying comments. PTAL.